api/src/main/java/com/codeforcommunity/rest/subrouter/ProtectedSiteRouter.java

```diff
@@ -420,7 +424,48 @@ private void handleApproveSiteImage(RoutingContext ctx) {
 
   private void registerGetUnapprovedImages(Router router) {
     Route unapprovedSiteImages = router.get("/unapproved_images");
-    unapprovedSiteImages.handler(this::handleGetUnapprovedImages);
+    unapprovedSiteImages.handler(this::handleFilterSiteImages);
+  }
+
+  private void handleFilterSiteImages(RoutingContext ctx) {
+    JWTData userData = ctx.get("jwt_data");
+
+    Optional<Timestamp> submittedStart =
+        RestFunctions.getOptionalQueryParam(ctx, "submittedStart", (date) -> new Timestamp(Date.valueOf(date).getTime()));
+    Optional<Timestamp> submittedEnd =
+        RestFunctions.getOptionalQueryParam(ctx, "submittedEnd", (date) -> new Timestamp(Date.valueOf(date).getTime()));
+    Optional<List<Integer>> siteIds =
+        RestFunctions.getOptionalQueryParam(
+            ctx,
+            "siteIds",
+            (string) ->
+                Arrays.stream(string.split(","))
+                    .map(Integer::parseInt)
+                    .collect(Collectors.toList()));
+    Optional<List<Integer>> neighborhoodIds =
```

Comment:
Optional since this might be the only case of it, but do you think it would make sense to abstract this repeated code out into a helper? I feel like parsing a list of optional query parameters would be quite a common pattern in the codebase  but I might be wrong. In the case that it is, we might be able to use generics for the parsing.